### PR TITLE
completing the patch article path and tests

### PR DESCRIPTION
**Pull request by libertyskies**
completing the patch article path, tests, and adding psql error handler controller

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -322,3 +322,117 @@ describe("POST /api/articles/:article_id/comments", () => {
...
+          author: "butter_bridge",
+          body: "I find this existence challenging",
+          created_at: "2020-07-09T20:11:00.000Z",
+          votes: 110,
+          article_img_url:
+            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+        });
+      });
+  });
+  test("returns an updated article when passed a valid object with a decrementing property", () => {
```
Comment (roisinoreilly) [inline __tests__/app.test.js:356]: nice edge case to test

File: controllers/errors.controller.js [hunk 2]
```diff
@@ -6,6 +6,12 @@ exports.handleCustomErrors = (err, req, res, next) => {
   }
 };
 
+exports.handlePSQLErrors = (err, req, res, next) => {
+  if (err.code === "22003") {
+    res.status(400).send({ msg: "Value out of accepted range" });
```
Comment (roisinoreilly) [inline controllers/errors.controller.js:11]: fab

File: models/articles.model.js [hunk 3]
```diff
@@ -62,3 +62,33 @@ exports.fetchArticle = async (sortby = "date", order = "DESC") => {
...
 };
+
+exports.updateArticle = async (inc_votes, id) => {
+  if (/\D/.test(id)) {
+    return Promise.reject({
+      status: 400,
+      msg: "Invalid ID type",
+    });
+  }
+  await checkValueExists("articles", "article_id", id);
```
Comment (roisinoreilly) [inline models/articles.model.js:73]: nicely done

**Review by roisinoreilly — APPROVED** [review]
really great work :) I'd just also like to see a test for if the inc_votes value isn't a number or if the object you're trying to send in the patch doesn't have an inc_votes key or other unnecessary keys. Once you've done that feel free to merge :)
